canonical/maas-ui
# feat(machines): Add button set on machine listing page MAASENG-1425

## Done

- Replace node action menu with node action button group on machine list
- Move machine list controls inside of header
- Add button to clear selection

## Todo

- [x] Fix cypress tests
- [x] Fix unit/integration tests
- [x] Add unit tests for button group (dis)appear functionality
- [x] Add unit test for clear selection button
- [x] Get advice on mobile styling

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to machine list
  - [x] Ensure machine list controls render correctly in header
- Select a machine / machines
  - [x] Ensure the filters, searchbox, and grouping select are replaced by the action buttons
- Dispatch some actions with selections
  - [x] Ensure the correct call is made for the corresponding actions and machines
- Click the "Clear selection" button
  - [x] Ensure all machines are deselected
  - [x] Ensure the action button group is replaced by the machine list controls

## Fixes

Fixes https://warthogs.atlassian.net/browse/MAASENG-1425

## Screenshots

### Before

![image](https://user-images.githubusercontent.com/35104482/226605301-63682a04-6628-4964-aa59-5638993c1776.png)

### After

**Without selection**
![image](https://user-images.githubusercontent.com/35104482/227494211-0a1486e2-0ad9-4111-aeba-a303921c1acd.png)

**With selection**
![image](https://user-images.githubusercontent.com/35104482/227494312-55d45c6d-e8db-4c8d-8c9f-085229b9c3f2.png)
